resteasy-core/src/main/java/org/jboss/resteasy/core/registry/MatchCache.java

```diff
@@ -26,7 +26,11 @@ public Key(final HttpRequest request, final int start) {
             this.path = start == 0 ? matchingPath : matchingPath.substring(start);
             this.start = start;
             this.method = request.getHttpMethod();
-            this.contentType = request.getHttpHeaders().getMediaType();
+            if (request.getHttpHeaders().getMediaType().getParameters().containsKey("boundary")) {
```

Comment:
is `request.getHttpHeaders().getMediaType()` always `not null` ? Just to be sure no NPE can jump from this.

Reply:
Actually the build is failing in resteasy-undertow testsuite because a NPE exactly there. So this has to be fixed @jimma .

Reply:
Fixed.